### PR TITLE
Do not use eval in minified piwik.js, replace JSON2 with JSON3

### LINKED ISSUE #5960: Do not use eval function in piwik.js (for CSP)
In order to implement CSP on my website, I disallowed `'unsafe-eval'` in `script-src` directive. But I cannot track visitors with Piwik, because piwik.js tracking code contains call to `eval` function twice.

Can you consider rewriting the code so it does not use `eval`? (Which is a bad idea nevertheless.)

### PR DESCRIPTION
**Pull request by tsteur**
fixes #5960  refs #1542

### CODE REVIEW

File: js/piwik.js [hunk 1]
```diff
@@ -5624,7 +6188,9 @@ if (typeof piwik_log !== 'function') {
 
         function getOption(optionName) {
             try {
-                return eval('piwik_' + optionName);
+                if (window['piwik_' + optionName]) {
```
Comment (tsteur) [inline js/piwik.js]: I'm not sure how the old tracker used to work but this should be the same

**Comment by tsteur** [comment]
After merging it, I'd run the tests on some older browsers via saucelabs on top to make sure it works with IE6 etc

File: tests/javascript/index.php [hunk 2]
```diff
@@ -439,10 +439,14 @@ function PiwikTest() {
         var src = '<?php
             $src = file_get_contents('../../js/piwik.js');
             $src = strtr($src, array('\\'=>'\\\\',"'"=>"\\'",'"'=>'\\"',"\r"=>'\\r',"\n"=>'\\n','</'=>'<\/'));
+            $src = substr($src, strpos($src, '/* startjslint */'));
```
Comment (tsteur) [inline tests/javascript/index.php]: This is a workaround to not run JSON3 through JSLINT as we cannot make it JSLINT compatible

File: js/piwik.js [hunk 3]
```diff
@@ -26,371 +26,933 @@
...
+} else {
+    (function () {
+        // we make sure to not break any site that uses JSON3 as well as we do not know if they run it in conflict mode
+        // or not.
+        var exports = {};
+
+        // Create a JSON object only if one does not already exist. We create the
+        // methods in a closure to avoid creating global variables.
+
+        /*! JSON v3.3.2 | http://bestiejs.github.io/json3 | Copyright 2012-2014, Kit Cambridge | http://kit.mit-license.org */
```
Comment (tsteur) [inline js/piwik.js]: BSD-3 Clause is compatible with MIT?

Comment (mattab) [inline js/piwik.js]: yes

**Comment by mattab** [comment]
Nice. Next step: https://github.com/piwik/piwik/issues/1542

**Comment by tsteur** [comment]
FYI: Ran JSON tests on IE6 and IE7 and worked. Not all other tests don't work because they are not compatible with older browsers

**Pull request merged**